File: src/test/java/Person.java
import com.google.cloud.firestore.annotation.Exclude;
import com.raylabz.firestorm.annotation.FirestormObject;
import com.raylabz.firestorm.annotation.ID;

@FirestormObject
public class Person {

    @ID private String id;
    private String firstName;
    private String lastName;
    private int age;
    private int ignoredField;

    public Person(String firstName, String lastName, int age, int ignoredField) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
        this.ignoredField = ignoredField;
    }

    private Person() {
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    @Exclude
    public int getIgnoredField() {
        return ignoredField;
    }

    public void setIgnoredField(int ignoredField) {
        this.ignoredField = ignoredField;
    }

    @Override
    public String toString() {
        return "Person{" +
                "id='" + id + '\'' +
                ", firstname='" + firstName + '\'' +
                ", lastname='" + lastName + '\'' +
                ", age=" + age +
                '}';
    }
}

File: flutter_test_app/lib/person.dart
import 'package:firestorm/annotations/firestorm_object.dart';

@FirestormObject()
class Person {

  String id;
  String firstname;
  String lastname;
  int age;
  double height;
  bool isEmployed;
  List<String> friends;

  Person(this.id, this.firstname, this.lastname, this.age, this.height,
      this.isEmployed, this.friends);

  @override
  String toString() {
    return 'Person{id: $id, firstname: $firstname, lastname: $lastname, age: $age, height: $height, isEmployed: $isEmployed, friends: $friends}';
  }



}